lib/msal-node/test/network/HttpClient.spec.ts

```diff
@@ -195,53 +210,52 @@ describe("HttpClient", () => {
-            (http.request as jest.Mock).mockImplementationOnce(mockHttpRequest(mockPostResponseBody, statusCodeOk, statusCodeOk));
+            (http.request as jest.Mock).mockImplementationOnce(mockHttpRequest(mockPostResponseBody, proxyStatusCodeOk, socketStatusCodeOk));
             await expect(httpClient.sendPostRequestAsync(url, postNetworkRequestOptionsWithProxyUrl, timeoutInMilliseconds)).rejects.toEqual(error);
         });
     });
 
-
     describe("Bad Status Code Error", () => {
-        const statusCodeError: number = 500;
-        const error: Error = new Error("Error connecting to proxy");
-        const networkResponse: NetworkResponse<MockedMetadataResponse> = getNetworkResponse<MockedMetadataResponse>(mockGetResponseBody, statusCodeError);
+        const proxyError: Error = new Error("Error connecting to proxy. Status Code: 500. Status Message: Unknown");
 
         describe("Get Request", () => {
-            test("Via Https", async () => {
-                (https.request as jest.Mock).mockImplementationOnce(mockHttpsRequest(mockGetResponseBodyBuffer, statusCodeError));
-                await expect(httpClient.sendGetRequestAsync(url)).resolves.toEqual(networkResponse);
+            test("Via Https - Server Error 500", async () => {
+                const serverErrorNetworkResponse: NetworkResponse<MockedMetadataResponse> = getNetworkResponse<MockedMetadataResponse>(mockServerErrorResponse, httpsStatusCodeFailure);
+                (https.request as jest.Mock).mockImplementationOnce(mockHttpsRequest(mockServerErrorResponseBodyBuffer, httpsStatusCodeFailure));
+                await expect(httpClient.sendGetRequestAsync(url)).resolves.toEqual(serverErrorNetworkResponse);
             });
 
             describe("Via Proxy", () => {
-                test("Http Status Code", async () => {
-                    (http.request as jest.Mock).mockImplementationOnce(mockHttpRequest(mockGetResponseBody, statusCodeError, statusCodeError));
-                    await expect(httpClient.sendGetRequestAsync(url, getNetworkRequestOptionsWithProxyUrl)).rejects.toEqual(expect.objectContaining(error));
+                test("Proxy Connection Status Code - Proxy Connection Error 500", async () => {
+                    (http.request as jest.Mock).mockImplementationOnce(mockHttpRequest(mockGetResponseBody, proxyStatusCodeFailure, httpsStatusCodeFailure));
+                    await expect(httpClient.sendGetRequestAsync(url, getNetworkRequestOptionsWithProxyUrl)).rejects.toEqual(proxyError);
                 });
 
-                test("Socket Status Code", async () => {
-                    (http.request as jest.Mock).mockImplementationOnce(mockHttpRequest(mockGetResponseBody, statusCodeOk, statusCodeError));
-                    await expect(httpClient.sendGetRequestAsync(url, getNetworkRequestOptionsWithProxyUrl)).rejects.toEqual(expect.objectContaining(error));
+                test("Socket (Http) Status Code - Server Error 500", async () => {
```

Comment:
Please add test that covers the ` error = "unknown_error";` branch, for example the server returns a 408 + empty body.

Reply:
After implementing your other feedback, I don't think this is necessary. Please take another look at the newest commit and let me know your thoughts.

Reply:
From a code coverage perspective, it would be good to cover the "unknwon_error" and the "client_error" cases - just add 2 simple unit tests for that. High code coverage is not necessarily a testament to well test code, but it's a good proxy for it and there's no effort (except copy-pasting 2 tests) to get that coverage.